src/info/info_command.c

```diff
@@ -85,24 +85,15 @@ static void renderIndexDefinitions(RedisModule_Reply *reply, IndexSpec *sp) {
   RedisModule_Reply_MapEnd(reply); // index_definition
 }
 
-/* FT.INFO {index}
- *  Provide info and stats about an index
- */
-int IndexInfoCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-  if (argc < 2) return RedisModule_WrongArity(ctx);
-
-  StrongRef ref = IndexSpec_LoadUnsafe(RedisModule_StringPtrLen(argv[1], NULL));
-  IndexSpec *sp = StrongRef_Get(ref);
-  if (!sp) {
-    return RedisModule_ReplyWithError(ctx, "Unknown index name");
-  }
-
-  RedisModule_Reply _reply = RedisModule_NewReply(ctx), *reply = &_reply;
-  bool has_map = RedisModule_HasMap(reply);
+void fillReplyWithIndexInfo(RedisSearchCtx* sctx, RedisModule_Reply *reply, bool obfuscate, bool withTimes) {
+  const bool has_map = RedisModule_HasMap(reply);
 
   RedisModule_Reply_Map(reply); // top
 
-  REPLY_KVSTR_SAFE("index_name", sp->name);
+  const IndexSpec *sp = sctx->spec;
+  IndexSpec *modifiableSpec = sctx->spec;
```

Comment:
A simple cast will do, no need to create to variables.

Reply:
I think the fact we modify the index state when reading the info is a problem.
Having the extra variable helps us to see which functions still need to modify the index state and hopefully modify them to remove that lack of const.

Reply:
Where do we modify it here?
I don't see any modifications.